rust/src/dataset/scanner.rs

```diff
@@ -667,6 +667,12 @@ impl Stream for DatasetRecordBatchStream {
     }
 }
 
+impl From<DatasetRecordBatchStream> for SendableRecordBatchStream {
```

Comment:
TBH, I'm not entirely clear why we have our own wrapper for `SendableRecordBatchStream`. Seems like it would be much less of a headache to just use the DataFusion type and make sure we provide a conversion trait for the errors.

Reply:
Lets just use Datafusion if feasible.